packages/lisk-p2p/src/p2p.ts

```diff
@@ -480,6 +497,61 @@ export class P2P extends EventEmitter {
 		return seedPeerUpdatedInfos;
 	}
 
+	private _pickRandomDiscoveredPeers(
+		count: number,
+	): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const discoveredPeerList: ReadonlyArray<P2PDiscoveredPeerInfo> = [
+			...this._newPeers.values(),
+			...this._triedPeers.values(),
+		]; // Peers whose values has been updated atleast once.
+
+		return selectRandomPeerSample(discoveredPeerList, count);
+	}
+
+	private _handleGetAllPeersRequest(request: P2PRequest): void {
+		// TODO later: Remove fields that are specific to the current Lisk protocol.
+		const protocolPeerInfoList: ProtocolPeerInfoList = {
+			success: true,
+			peers: this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE)
+				.map(
+					(peerInfo: P2PDiscoveredPeerInfo): ProtocolPeerInfo | undefined => {
+						const peerDiscoveredInfo = peerInfo;
+
+						if (!peerDiscoveredInfo) {
+							return undefined;
+						}
+
+						// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
+						return {
```

Comment:
This is not specific to this PR but this can be simplified as we discussed (and this object should also include custom properties).